content/integrations/maestro-integration.md

```diff
@@ -0,0 +1,95 @@
+2. Enter the desired **_Variable name_**, e.g. `MDEV_API_KEY`.
+3. Enter the desired variable value as **_Variable value_**.
+4. Enter the variable group name, e.g. **_maestro_**. Click the button to create the group.
+5. Make sure the **Secure** option is selected.
+6. Click the **Add** button to add the variable.
+
+7. Add the variable group to your `codemagic.yaml` file
+{{< highlight yaml "style=paraiso-dark">}}
+  environment:
+    groups:
+      - maestro
+{{< /highlight >}}
+
+## Managing Maestro flows
+After you have created your YAML tests flows inside the `.maestro` directory, you need to check the directory into your project repository.
+
+## Installing Maestro CLI
+Before you use maestro commands, you need first to simply install the CLI on the building machine using this command.
+{{< highlight yaml "style=paraiso-dark">}}
+scripts:
+    - name: Download Maestro
+      script: curl -Ls "https://get.maestro.mobile.dev" | bash
+{{< /highlight >}}
+
+## Uploading to Maestro Cloud
+
+First, you need to build your **Android (.apk) / iOS (.app)** apps, then use the `maestro cloud` command to test your app.
+
+{{< tabpane >}}
+{{< tab header="Android" >}}
+
+{{<markdown>}}
+See how to build your native android app [here](../yaml-quick-start/building-a-native-android-app/) or your Flutter app [here](../yaml-quick-start/building-a-flutter-app/).
+
+Add the following script to your `publishing` section:
+{{< highlight yaml "style=paraiso-dark">}}
+publishing:
+    scripts:
+    - name: Run tests on Maestro cloud
+        script: | 
+        export PATH="$PATH":"$HOME/.maestro/bin"
+        apkPath="/build/app/outputs/apk/release/app-release.apk"
+        maestro cloud \
+        --apiKey $MDEV_API_KEY \
+        $apkPath \
+        .maestro/
+{{< /highlight >}}
+
+Don't forget to change the value of the `apkPath` to your actual apk path.
+{{</markdown>}}
+{{< /tab >}}
+
+{{< tab header="iOS" >}}
+{{<markdown>}}
+For iOS, you need to upload your x86-compatible Simulator `.app` directory.
+
+Here's the script on how you can build it.
+{{< highlight yaml "style=paraiso-dark">}}
+  scripts:
+    - name: Build unsigned .app
+    script: | 
+        xcodebuild \
+        -workspace "ios/$XCODE_WORKSPACE" \
+        -scheme "$XCODE_SCHEME" \
+        -configuration "Debug" \
+        -sdk iphonesimulator \
+        -derivedDataPath ios/output
+{{< /highlight >}}
+
+Don't forget to add the environment variables that holds your XCode workspace name under `$XCODE_WORKSPACE` and the Scheme name under `$XCODE_SCHEME`, see the complete sample project from [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/tree/main/integrations/maestro_integration_demo_project/codemagic.yaml).
+{{</markdown>}}
+{{< /tab >}}
+{{< /tabpane >}}
+
+That's it, now if your Codemagic's build has failed at the maestro cloud step, then your tests has failed, or which means everything went well, and you can check out the build page for more details.
```

Comment:
```suggestion
If your Codemagic's build has failed at the Maestro cloud step, then your tests have failed. Otherwise, everything went well and you can check out the build page for more details.
```